File: python/2016/kickboxer-vengeance-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Kickboxer: Vengeance to the database
movies.insert(
    title="Kickboxer: Vengeance", 
    year=2016, 
    plot="A kick boxer is out to avenge his brother.", 
    rating=4.9
)

# Confirm that the movie was added
movie = movies.select(
    title="Kickboxer: Vengeance", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2016/kickboxer-vengeance-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Kickboxer: Vengeance" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Kickboxer: Vengeance",
                year: 2016,
                plot: "A kick boxer is out to avenge his brother.",
                rating: 4.9
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Kickboxer: Vengeance",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
